File: Backend/demo/src/main/java/com/example/paintapp/shapes/Square.java
package com.example.paintapp.shapes;

import com.example.paintapp.shapes.base.Shape;

public class Square extends Shape {
    public Square() {
        this.type = "Square";
    }

    //For Debugging
    @Override
    public void draw() {
        System.out.println("Drawing Square at (" + x1 + "," + y1 + ") with properties: " + properties);
    }
}

File: Frontend/src/app/models/square.ts
import { Shape } from './shape';

export class Square extends Shape {
  constructor(
    x1: number,
    y1: number,
    x2: number,
    y2: number,
    properties: { [key: string]: any } = {}
  ) {
    super('square', x1, y1, x2, y2, properties);
  }
}
